substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -392,100 +338,127 @@ impl PalletCmd {
 			for (s, selected_components) in all_components.iter().enumerate() {
 				// First we run a verification
 				if !self.no_verify {
+					let mut changes = genesis_changes.clone();
 					let state = &state_without_tracking;
-					let result = StateMachine::new(
-						state,
-						&mut changes,
-						&executor,
-						"Benchmark_dispatch_benchmark",
-						&(
-							&pallet,
-							&extrinsic,
-							&selected_components.clone(),
-							true, // run verification code
-							1,    // no need to do internal repeats
-						)
-							.encode(),
-						&mut extensions(),
-						&sp_state_machine::backend::BackendRuntimeCode::new(state)
-							.runtime_code()?,
-						CallContext::Offchain,
-					)
-					.execute()
-					.map_err(|e| {
-						format!("Error executing and verifying runtime benchmark: {}", e)
-					})?;
 					// Dont use these results since verification code will add overhead.
-					let _batch =
-						<std::result::Result<Vec<BenchmarkBatch>, String> as Decode>::decode(
-							&mut &result[..],
-						)
-						.map_err(|e| format!("Failed to decode benchmark results: {:?}", e))?
-						.map_err(|e| {
-							format!("Benchmark {pallet_name}::{extrinsic_name} failed: {e}",)
-						})?;
+					let _batch: Vec<BenchmarkBatch> = match Self::exec_state_machine::<
+						std::result::Result<Vec<BenchmarkBatch>, String>,
+						_,
+						_,
+					>(
+						StateMachine::new(
```

Comment:
I recall a helpder function like this to be useful: https://github.com/paritytech/try-runtime-cli/blob/main/core/src/lib.rs#L105, you can even specialize it further.

Reply:
Ah i see. The code is very similar.  But i also need to specify the initial `changes` overlay, to pass in the genesis state...  

A shared crate could still make sense, like `frame-runtime-interface`.